spec/features/audit_spec.rb

```diff
@@ -0,0 +1,239 @@
+    before do
+      sign_in(@user)
+    end
+
+    scenario "User can not visit the audits #index page" do
+      visit url_prefix + "/audits"
+      expect(page).to have_content("Access Denied")
+    end
+
+    scenario "User can not visit the audits #new page" do
+      visit url_prefix + "/audits/new"
+      expect(page).to have_content("Access Denied")
+    end
+
+    scenario "User can not visit the audits #edit page" do
+      visit url_prefix + "/audits/1/edit"
+      expect(page).to have_content("Access Denied")
+    end
+
+    scenario "User can not visit the audits #show page" do
+      visit url_prefix + "/audits/1"
+      expect(page).to have_content("Access Denied")
+    end
+  end
+
+  context "while signed in as an organization admin" do
+    before do
+      sign_in(@organization_admin)
+    end
+
+    scenario "User can filter the #index by storage location" do
+      storage_location = create(:storage_location, name: "here", organization: @organization)
+      storage_location2 = create(:storage_location, name: "there", organization: @organization)
+      create(:audit, organization: @organization, storage_location: storage_location)
+      create(:audit, organization: @organization, storage_location: storage_location2)
+
+      visit url_prefix + "/audits"
+      select storage_location.name, from: "filters_at_location"
+      click_button "Filter"
+
+      expect(page).to have_css("table tr", count: 2)
+    end
+
+    scenario "User can save progress of an audit" do
+      quantity = 7
+      storage_location = create(:storage_location, :with_items, organization: @organization)
+      visit url_prefix + "/audits"
+      click_link "New Audit"
+      select storage_location.name, from: "From storage location"
+      select Item.last.name, from: "audit_line_items_attributes_0_item_id"
+      fill_in "audit_line_items_attributes_0_quantity", with: quantity.to_s
+
+      expect do
+        click_button "Save Progress"
+        expect(Audit.last.in_progress?).to be_truthy
+        expect(Audit.last.line_items.count).to be(1)
+        expect(Audit.last.line_items.last.quantity).to be(7)
+      end.to change { Audit.count }.by(1)
+
+      expect(page).to have_content("Audit's progress was successfully saved.")
+      expect(page).to have_content(7)
+      expect(page).to have_content("In Progress")
+      expect(page).not_to have_content("Finalize Audit")
+      expect(page).to have_content("Resume Audit")
+      expect(page).to have_content("Delete Audit")
+    end
+
+    scenario "User can resume the audit that is in progress" do
+      item = create(:item)
+      storage_location = create(:storage_location, :with_items, item: item, item_quantity: 10)
+      audit = create(:audit,
+                     storage_location: storage_location,
+                     line_items_attributes: [
+                       { item_id: storage_location.items.first.id, quantity: 7 }
+                     ])
```

Comment:
So one thing that's cool is that since you defined a trait for `:with_items`, you can use that here! :)

Change this to: 
```ruby
audit = create(:audit, :with_items, storage_location: storage_location, item: storage_location.items.first, item_quantity: quantity
```
(again, using `quantity` instead of `7`, to avoid magic numbers)

You can do this same substitution on the other places where you have similar factory instantiations :+1: